scripts/rsfcreator.py

```diff
@@ -77,14 +75,20 @@ def rsf_for_line(line_dict, key_field, entry_type, key_prefix='', key=None):
             entry_type, key_prefix, key, timestamp, item_hash)
     return (item_line, entry_line)
 
+def merge_level_2(dct, other):
+    for k, v in other.items():
+        if (other.__contains__(k)):
+            dct[k] = {**dct[k], **other[k]}
+    return dct
+
 def generate_rsf(args):
     if args.register_data_root:
         register_def = read_register_from_local(args.phase, args.register_name, args.register_data_root)
         field_names = register_def['fields']
         fields_by_name = {fn: read_field_from_local(args.phase, fn, args.register_data_root) for fn in field_names}
         custom_fields_by_name = {fn: read_custom_field_from_local(args.phase, args.register_name, fn, args.register_data_root) for fn in field_names}
-        custom_fields_by_name = {k: v for k, v in custom_fields_by_name.items() if v}
-        fields_by_name = {**fields_by_name, **custom_fields_by_name}
+        custom_fields_by_name = remove_blanks(custom_fields_by_name)
```

Comment:
On second thought, perhaps the two expressions that generate `custom_fields_by_name` could be extracted as a function so the logic is isolated (and can be tested as well), something on the lines of:

```python
def custom_fields(phase, register_name, register_data_root, field_names):
    fields = {}
    for name in field_names:
        field = read_custom_field_from_local(phase, register_name, name, register_data_root)
        if field is not None:
            fields[name] = field

    return fields

def generate_rsf(args):
  ...
  custom_fields_by_name = custom_fields(args.phase, args.register_name,  args.register_data_root, field_names)
```